python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -455,11 +455,12 @@ def _doMatch(self, refCat, sourceCat, wcs, refFluxField, numUsableSources,
             if try_idx == 0 and \
                len(matcher_struct.matches) == 0 and \
                match_tolerance.lastMatchedPattern is not None:
-                # If we found a pattern on a previous run and can't
-                # find an optimistic match with the harshest tolerances,
-                # the match we found previously was likely bad. We append
-                # its index to a list so we do not use it in future match
-                # iterations.
+                # If we found a pattern on a previous match-fit interation and
+                # can't find an optimistic match on our first try with the 
+                # tolerances as found in the previous match-fit,
+                # the match we found in the last interation was likely bad. We
+                # append the bad match's index to the a list of
+                # patterns/matches to skip on subsiquent iterations.
```

Comment:
That's much clearer. Thanks! Two minor typos: "interation" -> "iteration" (twice) and "subsiquent" -> "subsequent".